packages/e2e-tests/test/e2e-direct.spec.ts

```diff
@@ -86,8 +86,10 @@ describe('e2e direct connection', function () {
         const shell = this.startTestShell({
           args: [await rs0.connectionString()],
         });
+        await shell.waitForPrompt();
```

Comment:
Pretty sure this is what causes failures like https://spruce.mongodb.com/task/mongosh_e2e_tests_macos_14_e2e_tests_darwin_patch_3c88417f1acaedf70850dfbc09675eb011a1f9db_692dafff12de3c00077a3431_25_12_01_15_10_57/logs?execution=0